### PR TITLE
Adopt PEP-621 for packaging

### PR DESCRIPTION
**Pull request by ssbarnea**
##### Contributor checklist

- [x] Provided the tests for the changes.
- [x] Assure PR title is short, clear, and good to be included in the user-oriented changelog

##### Maintainer checklist

- [x] Assure one of these labels is present: `backwards incompatible`, `feature`, `enhancement`, `deprecation`, `bug`, `dependency`, `docs` or `skip-changelog` as they determine changelog listing.
- [x] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

**Comment by ofek** [comment]
FYI it looks like this org wants to standardize on packaging and the first few migrated to Hatchling https://github.com/jazzband/help/issues/313

**Review by atugushev — APPROVED** [review]
LGTM 👍

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,119 @@
 [build-system]
-requires = ["setuptools>=45", "setuptools_scm[toml]>=6.2"]
+requires = ["setuptools>=63", "setuptools_scm[toml]>=7"]
 build-backend = "setuptools.build_meta"
 
+[project]
+# https://peps.python.org/pep-0621/#readme
+requires-python = ">=3.7"
```
Comment (atugushev) [inline pyproject.toml:7]: I wonder is there any decent `pyproject.toml` formatter?

Comment (ssbarnea) [reply pyproject.toml:7]: yep, prettier, i will configure it after this gets in.

Comment (ofek) [reply pyproject.toml:7]: https://github.com/tox-dev/pyproject-fmt

**Pull request merged**